modules/repository-s3/src/javaRestTest/java/org/elasticsearch/repositories/s3/AbstractRepositoryS3RestTestCase.java

```diff
@@ -0,0 +1,389 @@
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.repositories.s3;
+
+import io.netty.handler.codec.http.HttpMethod;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.cluster.routing.Murmur3HashFunction;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.junit.Before;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+public abstract class AbstractRepositoryS3RestTestCase extends ESRestTestCase {
+
+    public record TestRepository(String repositoryName, String clientName, String bucketName, String basePath) {
+
+        public Closeable register() throws IOException {
+            return register(UnaryOperator.identity());
+        }
+
+        public Closeable register(UnaryOperator<Settings> settingsUnaryOperator) throws IOException {
+            assertOK(client().performRequest(getRegisterRequest(settingsUnaryOperator)));
+            return () -> assertOK(client().performRequest(new Request("DELETE", "/_snapshot/" + repositoryName())));
+        }
+
+        private Request getRegisterRequest(UnaryOperator<Settings> settingsUnaryOperator) throws IOException {
+            return newXContentRequest(
+                HttpMethod.PUT,
+                "/_snapshot/" + repositoryName(),
+                (b, p) -> b.field("type", S3Repository.TYPE)
+                    .startObject("settings")
+                    .value(
+                        settingsUnaryOperator.apply(
+                            Settings.builder()
+                                .put("bucket", bucketName())
+                                .put("base_path", basePath())
+                                .put("client", clientName())
+                                .put("canned_acl", "private")
+                                .put("storage_class", "standard")
+                                .put("disable_chunked_encoding", randomBoolean())
+                                .build()
+                        )
+                    )
+                    .endObject()
+            );
+        }
+    }
+
+    protected abstract String getBucketName();
+
+    protected abstract String getBasePath();
+
+    protected abstract String getClientName();
+
+    protected static String getIdentifierPrefix(String testSuiteName) {
+        return testSuiteName + "-" + Integer.toString(Murmur3HashFunction.hash(testSuiteName + System.getProperty("tests.seed")), 16) + "-";
```

Comment:
This is in a method. I think we can just use the usual `randomIdentifier` and friends instead of performing hash manually?

Reply:
Unfortunately not, this is called when initializing some static fields in the test suite classes, which happens too early to use `randomIdentifier().